aws-datastore/src/androidTest/java/com/amplifyframework/datastore/storage/sqlite/StorageAdapterInstrumentedTestBase.java

```diff
@@ -0,0 +1,202 @@
+import com.amplifyframework.datastore.DataStoreException;
+import com.amplifyframework.datastore.storage.StorageItemChange;
+import com.amplifyframework.testmodels.commentsblog.AmplifyModelProvider;
+import com.amplifyframework.testutils.Await;
+
+import org.junit.After;
+import org.junit.Before;
+import org.junit.BeforeClass;
+
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Set;
+import java.util.concurrent.TimeUnit;
+
+import static org.junit.Assert.assertEquals;
+
+/**
+ * Test the functionality of {@link SQLiteStorageAdapter} operations.
+ */
+@SuppressWarnings("DesignForExtension") // Utility methods shouldn't be overwritten
+public abstract class StorageAdapterInstrumentedTestBase {
+    private static final long SQLITE_OPERATION_TIMEOUT_MS = TimeUnit.SECONDS.toMillis(2);
+    private static final String DATABASE_NAME = "AmplifyDatastore.db";
+
+    private SQLiteStorageAdapter sqliteStorageAdapter;
+    private Context context;
+
+    /**
+     * Enable strict mode for catching SQLite leaks.
+     */
+    @BeforeClass
+    public static void enableStrictMode() {
+        StrictMode.setVmPolicy(new StrictMode.VmPolicy.Builder()
+            .detectLeakedSqlLiteObjects()
+            .detectLeakedClosableObjects()
+            .penaltyLog()
+            .penaltyDeath()
+            .build());
+    }
+
+    /**
+     * Setup the required information for SQLiteStorageHelper construction.
+     * @throws DataStoreException If initialization of storage adapter fails
+     */
+    @Before
+    public void setUp() throws DataStoreException {
+        context = ApplicationProvider.getApplicationContext();
+        context.deleteDatabase(DATABASE_NAME);
+
+        ModelProvider modelProvider = AmplifyModelProvider.getInstance();
+        sqliteStorageAdapter = SQLiteStorageAdapter.forModels(modelProvider);
+        List<ModelSchema> setupResults = Await.result(
+            SQLITE_OPERATION_TIMEOUT_MS,
+            (Consumer<List<ModelSchema>> onResult, Consumer<DataStoreException> onError) ->
+                sqliteStorageAdapter.initialize(context, onResult, onError)
+        );
+
+        List<Class<? extends Model>> expectedModels = new ArrayList<>(modelProvider.models());
+        expectedModels.add(StorageItemChange.Record.class); // Internal
+        expectedModels.add(PersistentModelVersion.class); // Internal
+        assertEquals(expectedModels.size(), setupResults.size());
+    }
+
+    /**
+     * Drop all tables and database, terminate and delete the database.
+     * @throws DataStoreException from possible underlying DataStore exceptions
+     */
+    @After
+    public void tearDown() throws DataStoreException {
+        sqliteStorageAdapter.terminate();
+        context.deleteDatabase(DATABASE_NAME);
+    }
+
+    <T extends Model> void saveModel(@NonNull T model) throws DataStoreException {
```

Comment:
These methods should go in a class called `SynchronousStorageAdapter` that lives in the test sources. It would be analogous to the existing `SynchronousApi` and `SynchronousDataStore`, but it would pose a synchronous version of the storage adapter interface.

If you do it that way, then any test class may use this functionality, not just those who (must) extend this base class.